src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +32,14 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        // a/BLOCK, STREET, UNIT, POSTAL_CODE
+        String[] components = address.split(",");
+        Block block = new Block(components[BLOCK].substring(2, components[BLOCK].length()).trim());
```

Comment:
all your constants are initialized to zero above; how does the indexing work properly here?